File: Playlist#2/TokoFD/app/src/main/java/com/example/tokofd/MainActivity.java
package com.example.tokofd;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {

    private BottomNavigationView bottomNavigationView;

    // Map item ID ke Fragment
    private final Map<Integer, Fragment> fragmentMap = new HashMap<>();

    // Simpan fragment aktif
    private Fragment activeFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bottomNavigationView = findViewById(R.id.bottomNavigationView);

        // Inisialisasi map
        fragmentMap.put(R.id.nav_home, new HomeFragment());
        fragmentMap.put(R.id.nav_search, new SearchFragment());
        fragmentMap.put(R.id.nav_notifications, new NotificationsFragment());
        fragmentMap.put(R.id.nav_profile, new ProfileFragment());

        // Set fragment default
        if (savedInstanceState == null) {
            activeFragment = fragmentMap.get(R.id.nav_home);
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.nav_host_fragment, activeFragment)
                    .commit();
        }

        bottomNavigationView.setOnItemSelectedListener(item -> {
            int itemId = item.getItemId();

            // Tangani tombol cart sebagai pengecualian (karena activity)
            if (itemId == R.id.nav_cart) {
                startActivity(new Intent(MainActivity.this, CartActivity.class));
                return true;
            }

            Fragment selectedFragment = fragmentMap.get(itemId);

            if (selectedFragment != null && selectedFragment != activeFragment) {
                getSupportFragmentManager().beginTransaction()
                        .setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out)
                        .replace(R.id.nav_host_fragment, selectedFragment)
                        .commit();
                activeFragment = selectedFragment;
            }

            return true;
        });
    }
}

File: Playlist#2/SocialApp/app/src/main/java/com/example/socialapp/MainActivity.kt
package com.example.socialapp

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.socialapp.activities.CreatePostActivity
import com.example.socialapp.HomeFragment
import com.example.socialapp.ProfileFragment
import com.example.socialapp.SearchFragment
import com.example.socialapp.NotificationsFragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {

    private lateinit var bottomNav: BottomNavigationView
    private var lastSelectedItemId: Int = R.id.nav_home

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        bottomNav = findViewById(R.id.bottomNavigationView)

        // Default fragment saat pertama kali dibuka → Home
        if (savedInstanceState == null) {
            replaceFragment(HomeFragment())
            bottomNav.selectedItemId = R.id.nav_home
        }

        bottomNav.setOnItemSelectedListener { item ->
            when (item.itemId) {
                R.id.nav_home -> {
                    replaceFragment(HomeFragment())
                    lastSelectedItemId = item.itemId
                    true
                }
                R.id.nav_explore -> {
                    replaceFragment(SearchFragment())
                    lastSelectedItemId = item.itemId
                    true
                }
                R.id.nav_post -> {
                    startActivity(Intent(this, CreatePostActivity::class.java))
                    // Kembali ke item sebelumnya
                    bottomNav.selectedItemId = lastSelectedItemId
                    true
                }
                R.id.nav_notifications -> {
                    replaceFragment(NotificationsFragment())
                    lastSelectedItemId = item.itemId
                    true
                }
                R.id.nav_profile -> {
                    replaceFragment(ProfileFragment())
                    lastSelectedItemId = item.itemId
                    true
                }
                else -> false
            }
        }
    }

    private fun replaceFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragmentContainer, fragment)
            .commit()
    }
}
